CMakeLists.txt

```diff
@@ -94,7 +94,10 @@ else()
   endif()
 endif()
 message(STATUS "Building for architecture ARCH=${ARCH}")
-
+# Force C++17
+if(MSVC)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /std:c++17")
```

Comment:
We can not have that, this will break everybody.

Reply:
I'll try to find an alternative

Reply:
@marcalff The only other way I can get it to compile with vs2022 is to add "-DWITH_STL=CXX14"
Other wise i get

```
warning STL4038: The contents of <string_view> are available only with C++17 or later.
C:\Users\Administrator\Git\opentelemetry-cpp\api\include\opentelemetry/std/string_view.h(18): error C2039: 'string_view': is not a member of 'std'
```

This is during `stl` build, `nostd` works without this (of course).

Reply:
You can use `-DCMAKE_CXX_STANDARD=17` when running cmake command to use C++17 only.